pkg/util/mount/mount_helper_common.go

```diff
@@ -97,3 +98,18 @@ func PathExists(path string) (bool, error) {
 	}
 	return false, err
 }
+
+// isMountPointMatch returns true if the path in mp is the same as dir.
+// Handles case where mountpoint dir has been renamed due to stale NFS mount on
+// Linux.
+func isMountPointMatch(mp MountPoint, dir string) bool {
+	if mp.Path == dir {
+		return true
+	}
+	if runtime.GOOS == "linux" {
```

Comment:
This is a bit odd. My other thought was making IsNotMountPoint part of the interface.

Reply:
Sure, let me know how you want me to proceed...

Before I went this direction, I did a quick search of the codebase, and saw a lot of places in `pkg/kubelet` and `pkg/volume` that were doing switches on `runtime.GOOS`, so I kind of ran with that precedent. :) It made for more compact code that could just be one common internal function.

Reply:
Let's get @saad-ali 's thoughts on this.  I think it's a little odd because we already have os-specific implementations of the mounter.

Reply:
Are there instances where this code wouldn't be run on Linux, and would that be a bug? I would prefer to err on the side of safety if the mounter code somehow winds up as the default code path.

Reply:
I agree with the higher level goal here which is remove `IsMountPointMatch(...)` from the Mount interface since it is an implementation detail for Mount.

There are already `mount_helper_windows.go` and `mount_helper_unix.go` files, why not make `isMountPointMatch(...)` arch specific, and put a definition in each of those files?

Reply:
I had tried that approach originally, but was running into issues because I thought I needed an  `unsupported` implementation along with Linux and Windows. I had failed to notice that the `mount_helper_unix.go` file is actually `+build !windows`, meaning that it builds on Darwin as well. Therefore anything in there is actually for *both* Linux and Darwin. Before I was getting a redeclaration error.

I've fixed this. Should be good to review again.

Reply:
> I had failed to notice that the mount_helper_unix.go file is actually +build !windows, meaning that it builds on Darwin as well. Therefore anything in there is actually

Unrelated to this PR, but why are we doing `+build !windows` rather then `+build linux`?